### PR TITLE
Added webapp-with-golang

### PR DESCRIPTION
**Pull request by rprakashg**
This template can be used to create web applications developed in GoLang on Azure. Template is originally created by Wade Wegner of microsoft.

### CODE REVIEW

**Comment by singhkay** [comment]
@rprakashg Can you please sign the Contribution Licence Agreement before we can accept this PR? (All electronic, no faxing!)

https://cla.azure.com 

Thanks!

**Comment by azurecla** [comment]
__@rprakashg__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

**Comment by singhkay** [comment]
@rprakashg Thank for signing the CLA! 

I was chatting with @davidebbo from the Websites team and he suggested we add a 101 to the folder name to indicate this is a beginner template. So let's rename the folder to "101-webapp-with-golang"

@davidebbo will be suggesting additional best practices/guidance around Websites

**Comment by davidebbo** [comment]
You can look at [this file](https://github.com/davidebbo/AzureWebsitesSamples/blob/master/ARMTemplates/WebAppManyFeatures.json) for a sample template using the latest version.

**Comment by singhkay** [comment]
@rprakashg Can you looking into updating the template to use latest API headers?

**Comment by rprakashg** [comment]
Yes I'm working on this, did add a 101 as folder prefix and updating the template json per David Ebbo's reference template. I guess sSince the folder was renamed I would need to close this PR and send a new one right?

**Comment by singhkay** [comment]
@rprakashg Thanks! Just a commit to this PR should work as well

**Comment by rprakashg** [comment]
@singhkay done

File: 101-webapp-with-golang/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                    "apiVersion": "2015-06-01",
+                    "name": "web",
+                    "type": "config",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Web/Sites', variables('goSite'))]"
+                    ],
+                    "properties": {
+                        "phpVersion": "off",
+                        "netFrameworkVersion": "v4.5",
+                        "use32BitWorkerProcess": "[parameters('use32bitWorkerProcess')", /* 64-bit platform */
```
Comment (singhkay) [inline 101-webapp-with-golang/azuredeploy.json]: Can you remove the comments? the CLI will fail while trying to deploy this template due to comments

**Comment by singhkay** [comment]
@rprakashg Please remove the comments
@davidebbo Can you take a look and approve?

**Pull request merged**